runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/ExecutableStageDoFnOperator.java

```diff
@@ -502,6 +692,10 @@ public void close() throws Exception {
     processWatermark1(Watermark.MAX_WATERMARK);
     while (getCurrentOutputWatermark() < Watermark.MAX_WATERMARK.getTimestamp()) {
       invokeFinishBundle();
+      if (hasSdfProcessFn) {
+        // Sleep for 5s to wait for any SDF timer to be fired.
+        Thread.sleep(5000);
```

Comment:
Perhaps add a `TODO` here, since we may want to eventually get rid of the sleep?

Reply:
Thanks! Filed here: https://issues.apache.org/jira/browse/BEAM-11210